aliyun/dns/connection.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding:utf-8 -*-
+# Copyright 2014, Quixey Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import json
+import time
+import datetime
+import logging
+
+import dateutil.parser
+
+from aliyun.connection import Connection
+
+BLOCK_TILL_RUNNING_SECS = 600
+
+logger = logging.getLogger(__name__)
+
+class Error(Exception):
+
+    """Base exception class for this module."""
+
+
+class DnsConnection(Connection):
+    """A connection to Aliyun DNS service.
+
+    Args:
+        region_id (str): NOT IN USE FOR DNS
+                         But needed for backward comptibility
+                         The id of the region to connect to.
+        access_key_id (str): The access key id.
+        secret_access_key (str): The secret access key.
+    """
+    def __init__(self, region_id='cn-hangzhou', access_key_id=None, secret_access_key=None):
+        super(DnsConnection, self).__init__(
+	    region_id, 'dns', access_key_id=access_key_id,
+	    secret_access_key=secret_access_key)
+    
+    def add_record(self, rr=None, type='A', value=None, domainname="quixey.be"):
+        """
+	Add a DNS record to specified Domain.
+
+	Args:
+	    rr (str): Resource Record to add.
+	    value (str): The IP address of the RR.
+	    type (str): Type of the Resource Record, A, PTR, CNAME for instance.
+	    domainname (str): The domain name the rr will be added into.
+	
+	Returns: 
+	"""
+
+	if rr is None or value is None:
+	    exit("ERROR: Both RR and its value MUST be supplied.")
```

Comment:
I don't think exit is defined. In any case, in a library you should raise an exception in case of errors, and typically you'd raise ValueError if parameters to methods were incorrect.